### PR TITLE
Fix to resolve BZ2589

### PR DESCRIPTION
**Pull request by DmitryLitvintsev**
This should address BZ2589. Not tested.

### CODE REVIEW

File: src/configuration_server.py [hunk 1]
```diff
@@ -881,14 +881,21 @@ def get_media_changer(self, ticket):
...
             if ret:
-                break
+                # handle TFF1-AB variety
+                mc = self.get_dict_entry(ret)
+                if mc.get('remote_media_changer'):
+                    ret = mc.get('remote_media_changer')
+                    if ret:
+                        break
+                else:
+                    break
```
Comment (renbauer-fnal) [inline src/configuration_server.py]: 838-898 are equivalent to simply:
```
if mc.get('remote_media_changer'):
    ret = mc.get('remote_media_changer')
break
```
Correct? I think that is more clear

**Review by renbauer-fnal — CHANGES_REQUESTED** [review]

File: src/configuration_server.py (hunk 1 above)
Comment (DmitryLitvintsev) [reply src/configuration_server.py]: OK

**Review by renbauer-fnal — APPROVED** [review]

**Comment by DmitryLitvintsev** [comment]
Just a note here. This needs to be tested. But I do not think dmsen setup has these A/B "sub-changers"

**Comment by DmitryLitvintsev** [comment]
OK, I tested it in a roundabout way..... Just testing relevant call in volume_clerk and mimicking MC "substitution". It seems to work:

```
[enstore@enstore00 ~]$ python test.py 
('replace ', 'TFF1_B.media_changer', ' with TFF1.media_changer')
('Volume state is ', 'O')
('Media changer is ', 'TFF1_B.media_changer')
unable to get volume state from TFF1_B.media_changer: TIMEDOUT: TFF1_B.media_changer
('Volume state is ', 'unknown')
[enstore@enstore00 ~]$ 
```
`test.py` looks like:

```
import enstore_functions2
import configuration_client
import e_errors

class Dummy:

    def __init__(self):
        self.csc   = configuration_client.ConfigurationClient((enstore_functions2.default_host(),
                                                               enstore_functions2.default_port()))
    def get_media_changer_state(self, lib, volume, m_type, overwrite=False):

        if not lib.split('.')[0] in self.csc.get_library_managers().keys():
            return "no_lib"  #Not a fatal error.                                

        if len(lib) < 16 or lib[-16:] != '.library_manager':
            lib = lib + '.library_manager'
        m_changer = self.csc.get_media_changer(lib)
        if not m_changer:
            return "no_mc"  #Not a fatal error.                                 

        if overwrite:
            print ("replace ", m_changer, " with TFF1.media_changer")
            m_changer = "TFF1.media_changer"
        else:
            print("Media changer is ", m_changer)
            
        
        import media_changer_client
        mcc = media_changer_client.MediaChangerClient(self.csc, m_changer )
        reply_ticket = mcc.viewvol(volume, m_type, rcv_timeout = 5,
                                   rcv_tries = 5)
        if not e_errors.is_ok(reply_ticket):
            print("unable to get volume state from %s: %s: %s" % \
                  (m_changer, reply_ticket['status'][0], \
                   reply_ticket['status'][1]))
            return "unknown"  #Fatal error.                                     

        #Most common values are:                                                
        # "O" for occupied in its home slot                                     
        # "M" for mounted in drive                                              
        # "E" for ejected                                                       
        # "U" for undefined/unknown                                             
        stat = reply_ticket['status'][3]

        return stat

if __name__ == "__main__":
    volume = "FA0156L8"
    library = "TFF1-LTO8"
    media_type = "LTO8"
    dummy = Dummy()
    ret = dummy.get_media_changer_state(library, volume, media_type, True)
    print("Volume state is ", ret)
    ret = dummy.get_media_changer_state(library, volume, media_type)
    print("Volume state is ", ret)

``` 

(it is available at enstore@enstore00) 

as you can see first call where I replace  'TFF1_B.media_changer', ' with TFF1.media_changer' works and returns 'O' status and 
second call where I do not replace 'TFF1_B.media_changer', ' with TFF1.media_changer' fails with timeout. 

Big question here - is this a normal behavior or not I cannot tell. But replacing "B" with that "remote_media_changer" gets things 
going. So hence this patch.

**Pull request merged**